File: VehicleStore/src/VehicleStore.Infrastructure/Data/Mappers/VehicleMapper.cs
using VehicleStore.Domain.Entities;
using VehicleStore.Infrastructure.Data.Records;

namespace VehicleStore.Infrastructure.Data.Mappers;

internal static class VehicleMapper
{
    public static Vehicle ToDomain(VehicleRecord record) => new()
    {
        Id = record.Id,
        PlateNumber = record.PlateNumber,
        Document = record.Document,
        Brand = record.Brand,
        Model = record.Model,
        Year = record.Year,
        Price = record.Price,
        IsSold = record.IsSold,
        CreatedAt = record.CreatedAt,
        UpdatedAt = record.UpdatedAt
    };

    public static VehicleRecord ToRecord(Vehicle vehicle) => new()
    {
        Id = vehicle.Id,
        PlateNumber = vehicle.PlateNumber,
        Document = vehicle.Document,
        Brand = vehicle.Brand,
        Model = vehicle.Model,
        Year = vehicle.Year,
        Price = vehicle.Price,
        IsSold = vehicle.IsSold,
        CreatedAt = vehicle.CreatedAt,
        UpdatedAt = vehicle.UpdatedAt
    };
}

File: vehicle-store-web/src/app/data/mappers/vehicle.mapper.ts
import { Vehicle } from '../../domain/models/vehicle.model';
import { VehicleResponse } from '../../domain/interfaces/responses/vehicle.response';

export class VehicleMapper {
  static toDomain(response: VehicleResponse): Vehicle {
    return {
      id: response.id,
      plateNumber: response.plateNumber,
      document: response.document,
      brand: response.brand,
      model: response.model,
      year: response.year,
      price: response.price,
      isSold: response.isSold,
      createdAt: response.createdAt,
      updatedAt: response.updatedAt
    };
  }

  static toDomainList(responses: VehicleResponse[]): Vehicle[] {
    return responses.map(VehicleMapper.toDomain);
  }
}
